scripts/install_KLU_Sundials.py

```diff
@@ -59,132 +282,55 @@ def download_extract_library(url, download_dir):
-        OpenMP_C_FLAGS = "-Xpreprocessor -fopenmp -I/usr/local/opt/libomp/include"
-        OpenMP_CXX_FLAGS = "-Xpreprocessor -fopenmp -I/usr/local/opt/libomp/include"
-        OpenMP_C_LIB_NAMES = "omp"
-        OpenMP_CXX_LIB_NAMES = "omp"
-        OpenMP_omp_LIBRARY = "/usr/local/opt/libomp/lib/libomp.dylib"
-
-    cmake_args += [
-        "-DLDFLAGS=" + LDFLAGS,
-        "-DCPPFLAGS=" + CPPFLAGS,
-        "-DOpenMP_C_FLAGS=" + OpenMP_C_FLAGS,
-        "-DOpenMP_C_LIB_NAMES=" + OpenMP_C_LIB_NAMES,
-        "-DOpenMP_omp_LIBRARY=" + OpenMP_omp_LIBRARY,
-    ]
-
-# SUNDIALS are built within download_dir 'build_sundials' in the PyBaMM root
-# download_dir
-build_dir = os.path.abspath(os.path.join(download_dir, "build_sundials"))
-if not os.path.exists(build_dir):
-    print("\n-" * 10, "Creating build dir", "-" * 40)
-    os.makedirs(build_dir)
-
-sundials_src = f"../sundials-{sundials_version}"
-print("-" * 10, "Running CMake prepare", "-" * 40)
-subprocess.run(["cmake", sundials_src, *cmake_args], cwd=build_dir, check=True)
+if args.force:
+    print(
+        "The '--force' option is activated: installation will be forced, ignoring any existing libraries."
+    )
+    safe_remove_dir(os.path.join(download_dir, "build_sundials"))
+    safe_remove_dir(os.path.join(download_dir, f"SuiteSparse-{SUITESPARSE_VERSION}"))
+    safe_remove_dir(os.path.join(download_dir, f"sundials-{SUNDIALS_VERSION}"))
+    sundials_found, suitesparse_found = False, False
+else:
+    # Check whether the libraries are installed
+    sundials_found, suitesparse_found = check_libraries_installed(install_dir)
 
-print("-" * 10, "Building the sundials", "-" * 40)
-make_cmd = ["make", f"-j{cpu_count()}", "install"]
-subprocess.run(make_cmd, cwd=build_dir, check=True)
+if __name__ == "__main__":
+    # Determine which libraries to download based on whether they were found
+    if not sundials_found and not suitesparse_found:
+        # Both SUNDIALS and SuiteSparse are missing, download and install both
+        parallel_download(
+            [
+                (SUITESPARSE_URL, SUITESPARSE_CHECKSUM),
+                (SUNDIALS_URL, SUNDIALS_CHECKSUM),
+            ],
+            download_dir,
+        )
+        install_suitesparse(download_dir)
+        install_sundials(download_dir, install_dir)
+    elif not sundials_found and suitesparse_found:
+        # Only SUNDIALS is missing, download and install it
+        parallel_download([(SUNDIALS_URL, SUNDIALS_CHECKSUM)], download_dir)
+        install_sundials(download_dir, install_dir)
+    elif sundials_found and not suitesparse_found:
+        # Only SuiteSparse is missing, download and install it
+        parallel_download([(SUITESPARSE_URL, SUITESPARSE_CHECKSUM)], download_dir)
+        install_suitesparse(download_dir)
+    else:
+        # Both libraries are found and no force installation is requested
+        print("Both SUNDIALS and SuiteSparse libraries are already installed.")
```

Comment:
For this case, you might be able to do
```suggestion
    else:
        if not sundials_found:
            parallel_download([(SUNDIALS_URL, SUNDIALS_CHECKSUM)], download_dir)
            install_sundials(download_dir, install_dir)
        if not suitesparse_found:
            parallel_download([(SUITESPARSE_URL, SUITESPARSE_CHECKSUM)], download_dir)
            install_suitesparse(download_dir)
```
This stops you from doing additional checks gainst the other libraries. A small optimization, but it makes the code much simpler.

Reply:
@AlessioBugetti This should be my last finding

Reply:
Thank you for your feedback and observations!

Commit [here](https://github.com/pybamm-team/PyBaMM/pull/3719/commits/63a4ed3e7be0fc9039af2b6f0001c0ffdf70bb5e)